### ISSUE TITLE
Better handling of imagefactory connection error

### ISSUE THREAD

**Issue by jprovaznik** [aeolus-incubator/tim#83]
If imagefactory connection fails, this error is raised in create_factory_provider_image and create_factory_target_image:

Connection refused - connect(2)
/usr/share/ruby/net/http.rb:762:in `initialize'
 /usr/share/ruby/net/http.rb:762:in`open'
 /usr/share/ruby/net/http.rb:762:in `block in connect'
 /usr/share/ruby/timeout.rb:54:in`timeout'
 /usr/share/ruby/timeout.rb:99:in `timeout'
 /usr/share/ruby/net/http.rb:762:in`connect'
 /usr/share/ruby/net/http.rb:755:in `do_start'
 /usr/share/ruby/net/http.rb:744:in`start'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/webmock-1.8.7/lib/webmock/http_lib_adapters/net_http.rb:90:in `request_with_webmock'
 /usr/share/ruby/net/http.rb:1307:in`send_entity'
 /usr/share/ruby/net/http.rb:1096:in `post'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/connection.rb:113:in`block in request'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activesupport-3.2.8/lib/active_support/notifications.rb:123:in `block in instrument'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activesupport-3.2.8/lib/active_support/notifications/instrumenter.rb:20:in`instrument'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activesupport-3.2.8/lib/active_support/notifications.rb:123:in `instrument'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/connection.rb:110:in`request'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/bundler/gems/aeolus-image-rubygem-4bffe3f60ce3/lib/aeolus_image/active_resource_oauth_client.rb:30:in `request_with_oauth'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/connection.rb:98:in`block in post'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/connection.rb:218:in `with_auth'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/connection.rb:98:in`post'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/base.rb:1360:in `create'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/observing.rb:19:in`create_with_notifications'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/base.rb:1155:in `save'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/validations.rb:79:in`save_with_validation'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/observing.rb:19:in `save_with_notifications'
 /home/jprovazn/devel/bundle-conductor/ruby/1.9.1/gems/activeresource-3.2.8/lib/active_resource/base.rb:1172:in`save!'
 /home/jprovazn/devel/tim/app/models/tim/provider_image.rb:47:in `create_factory_provider_image'

More descriptive error instead of "Connection refused - connect(2)" should be returned.
I think this should be fixed before 2.0 release.

**Comment by jguiditta** [comment]
Just to clarify here, the december release is NOT '2.0', that is a downstream release number, and we will continue fixing these bugs as they come up before that release goes out.

**Comment by mtaylor** [comment]
Since Jan was not able to tag issues I was trying to find out which ones were blockers.  From our conversation this morning: 

(10:15:46 AM) jprovazn: martyntaylor: I've created some issues in last days, if some of them is blocker, it's mentioned in description
(10:17:20 AM) martyntaylor: jprovazn: I suppose: I think this should be fixed before 2.0 release. == blocker?
(10:17:40 AM) jprovazn: right ;)
(10:17:51 AM) jprovazn: at least IMO

So I fixed it.
